sessiontxn/staleread/processor.go

```diff
@@ -172,14 +187,19 @@ func (p *staleReadProcessor) OnSelectTable(tn *ast.TableName) error {
 	return p.evaluateFromStmtTSOrSysVariable(stmtAsOfTS)
 }
 
-func (p *staleReadProcessor) OnExecutePreparedStmt(preparedTSEvaluator StalenessTSEvaluator) (err error) {
+func (p *staleReadProcessor) OnExecutePreparedStmt(node ast.Node, preparedTSEvaluator StalenessTSEvaluator) (err error) {
 	if p.evaluated {
 		return errors.New("already evaluated")
 	}
 
 	if p.sctx.GetSessionVars().InTxn() {
-		if preparedTSEvaluator != nil {
-			return errAsOf.FastGenWithCause("as of timestamp can't be set in transaction.")
+		// allow the stale-read evaluator inherit from the variables, just ignore them
+		if stmt, ok := node.(*ast.SelectStmt); ok {
```

Comment:
why we need this change?

Reply:
No, we don't need :cry: . I removed this change.